packages/selection/src/containers/SelectionContainer.spec.js

```diff
@@ -0,0 +1,493 @@
+            expect(items.at(2)).toHaveProp('data-focused', false);
+          });
+        });
+      });
+    });
+  });
+
+  describe('getItemProps', () => {
+    it('throws if no item is applied', () => {
+      console.error = jest.fn(); // eslint-disable-line no-console
+
+      expect(() => {
+        mount(
+          <SelectionContainer>
+            {({ getContainerProps, getItemProps }) => (
+              <div {...getContainerProps()}>
+                <div {...getItemProps()}>Example</div>
+              </div>
+            )}
+          </SelectionContainer>
+        );
+      }).toThrow('Accessibility Error: You must provide an "item" option to "getItemProps()"');
+    });
+
+    it('throws if no focusRef prop is applied', () => {
+      console.error = jest.fn(); // eslint-disable-line no-console
+
+      expect(() => {
+        mount(
+          <SelectionContainer>
+            {({ getContainerProps, getItemProps }) => (
+              <div {...getContainerProps()}>
+                <div {...getItemProps({ item: 'example' })}>Example</div>
+              </div>
+            )}
+          </SelectionContainer>
+        );
+      }).toThrow('Accessibility Error: You must provide a "focusRef" option to "getItemProps()"');
+    });
+
+    it('does not throw if item and focusRef props are applied', () => {
+      expect(() => {
+        mount(
+          <SelectionContainer>
+            {({ getContainerProps, getItemProps }) => (
+              <div {...getContainerProps()}>
+                <div {...getItemProps({ item: 'example', focusRef: {} })}>Example</div>
+              </div>
+            )}
+          </SelectionContainer>
+        );
+      }).not.toThrow();
+    });
+
+    it('applies accessibility role attribute', () => {
+      expect(findItems(wrapper).first()).toHaveProp('role', 'option');
+    });
+
+    it('applies default selected aria value if none provided', () => {
+      expect(findItems(wrapper).first()).toHaveProp('aria-selected');
+    });
+
+    it('applies custom selected aria value if provided', () => {
+      wrapper = mount(<BasicExample selectedAriaKey="aria-pressed" />);
+      expect(findItems(wrapper).first()).toHaveProp('aria-pressed');
+    });
+
+    describe('onClick', () => {
+      it('should select item that was clicked', () => {
+        findItems(wrapper)
+          .last()
+          .simulate('click');
+        expect(findItems(wrapper).last()).toHaveProp('aria-selected', true);
+      });
+
+      it('should remove focus from all items', () => {
+        findItems(wrapper)
+          .last()
+          .simulate('click');
+
+        const items = findItems(wrapper);
+
+        expect(items.at(0)).toHaveProp('data-focused', false);
+        expect(items.at(1)).toHaveProp('data-focused', false);
+        expect(items.at(2)).toHaveProp('data-focused', false);
+      });
+    });
+
+    // TODO: Do we want to support this?
+    describe.skip('custom index ordering', () => {
```

Comment:
Was this ever used on the render-prop container?

Reply:
Haha nope. Lets remove this.